pkg/controller/namespace/namespace_controller.go

```diff
@@ -132,7 +132,9 @@ func (nm *NamespaceController) enqueueNamespace(obj interface{}) {
 		glog.Errorf("Couldn't get key for object %+v: %v", obj, err)
 		return
 	}
-	nm.queue.Add(key)
+	// delay processing namespace deletions long enough for the lifecycle admission
+	// plugins to also observe the deletion and stop new objects from being created
+	nm.queue.AddAfter(key, 5*time.Second)
```

Comment:
I'm not sure #22007 helps in a case where you have namespaces stored in one etcd, and namespaced objects stored in a different etcd served by different API servers. There's not necessarily any coherence between the namespace's resourceVersion and the resourceVersion of an object in that namespace